src/ts/components/listbox.ts

```diff
@@ -0,0 +1,119 @@
+import {Button, ButtonConfig} from './button';
+import {ListSelector, ListSelectorConfig} from './listselector';
+import {DOM} from '../dom';
+
+/**
+ * A element to select a single item out of a list of available items.
+ *
+ * DOM example:
+ * <code>
+ *   <div class='ui-listbox>
+ *     <div class='ui-listboxitem-wrapper>
+ *       <div class='ui-listboxitem'>
+ *         <button class='ui-listboxitem-button'>label</button>
+ *       </div>
+ *     </div>
+ *     ...
+ *   </div
+ * </code>
+ */
+export class ListBox extends ListSelector<ListSelectorConfig> {
+  private listBoxElement: DOM;
+
+  private static readonly SELECTED_LIST_BOX_ITEM_CLASS = 'ui-listboxitem-wrapper-selected';
+  private static readonly LIST_BOX_ITEM_WRAPPER_CLASS = 'ui-listboxitem-wrapper';
+
+  constructor(config: ListSelectorConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-listbox',
+    } as ListSelectorConfig, this.config);
+  }
+
+  protected toDomElement(): DOM {
+    let listBoxElement = new DOM('div', {
+      'id': this.config.id,
+      'class': this.getCssClasses(),
+    });
+
+    this.listBoxElement = listBoxElement;
+    this.updateDomItems();
+
+    return listBoxElement;
+  }
+
+  protected updateDomItems(selectedValue: string = null) {
+    // Delete all children
+    this.listBoxElement.empty();
```

Comment:
Use `this.getDomElement()` instead of `tihs.listBoxElement`

Reply:
This will result in an infinite loop because the `updateDomItems` are called from the `getDomElement` function to handle initial state configuration.

Similar to the `SelectBox`

Reply:
Ok then it's fine :)